docs/2-building-blocks-of-oop-part-1/hometask/src/Shape.ts

```diff
@@ -1,3 +1,45 @@
+import { Point } from "./Point";
+
 export abstract class Shape {
+  protected color: string;
+  protected filled: boolean;
+  protected points: Point[];
+
+  constructor(points: Point[]);
+  constructor(points: Point[], color: string, filled: boolean);
+  constructor(...args: any[]) {
+    const [points, color, filled] = args;
+    if (points.length <= 2) {
+      throw new Error("Shape should contain at least 3 points");
+    }
+    if (args.length === 1) {
+      this.color = "green";
+      this.filled = true;
+      this.points = points;
+    }
+    if (args.length === 3) {
+      this.color = color;
+      this.filled = filled;
+      this.points = points;
+    }
+  }
+
   abstract getType(): string;
+
+  toString(): string {
+    const isFilled = this.filled ? "filled" : "not filled";
+    const pointsAsString = this.points.map((point: Point) => point.toString());
+    return `A Shape with color of ${this.color} and ${isFilled}. Points: ${pointsAsString.join(", ")}.`;
```

Comment:
there is no need to use variable `pointsAsString`

the current line can be updated:
```
return `A Shape with color of ${this.color} and ${isFilled}. Points: ${this.points.join(", ")}.`;
```